### PR TITLE
nxrecorder: add support to record media file.

### PR DESCRIPTION
**Pull request by qiaohaijiao**
## Summary

nxrecorder:  add support to record media file.

## Impact

New functionality

## Testing

Xiaomi Vela

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: system/nxrecorder/Kconfig [hunk 1]
```diff
@@ -43,5 +43,15 @@ config NXRECORDER_INCLUDE_HELP
...
 
+config NXRECORDER_FMT_FROM_EXT
+    bool "Include code to determine Audio format from extension"
+    default y
+    ---help---
+        Compiles in extra code to determine audio format based
+        on the filename extension for known file types.
+        This feature is used if the format is not manually
+        specified, and will take priority over the more lengthy
+        file content detection approach.
```
Comment (pkarashchenko) [inline system/nxrecorder/Kconfig:54]: Let's use TABs in Kconfig

File: system/nxrecorder/nxrecorder.c [hunk 2]
```diff
@@ -101,7 +162,79 @@ static int nxrecorder_opendevice(FAR struct nxrecorder_s *precorder)
           return -ENOENT;
         }
 
-      return OK;
+      cap.ac_len = sizeof(cap);
+      cap.ac_type    = AUDIO_TYPE_QUERY;
+      cap.ac_subtype = AUDIO_TYPE_QUERY;
+
+      if (ioctl(precorder->dev_fd, AUDIOIOC_GETCAPS,
+               (unsigned long)&cap) == cap.ac_len)
```
Comment (pkarashchenko) [inline system/nxrecorder/nxrecorder.c:170]:
```suggestion
                (uintptr_t)&cap) == cap.ac_len)
```

File: system/nxrecorder/nxrecorder.c [hunk 3]
```diff
@@ -101,7 +162,79 @@ static int nxrecorder_opendevice(FAR struct nxrecorder_s *precorder)
           return -ENOENT;
         }
 
-      return OK;
+      cap.ac_len = sizeof(cap);
+      cap.ac_type    = AUDIO_TYPE_QUERY;
+      cap.ac_subtype = AUDIO_TYPE_QUERY;
```
Comment (pkarashchenko) [inline system/nxrecorder/nxrecorder.c:167]:
```suggestion
      cap.ac_len     = sizeof(cap);
      cap.ac_type    = AUDIO_TYPE_QUERY;
      cap.ac_subtype = AUDIO_TYPE_QUERY;
```

File: system/nxrecorder/nxrecorder.c [hunk 4]
```diff
@@ -101,7 +162,79 @@ static int nxrecorder_opendevice(FAR struct nxrecorder_s *precorder)
...
+                {
+                  /* Prepare to get sub-formats for
+                   * this main format
+                   */
+
+                  cap.ac_subtype = format;
+                  cap.ac_format.b[0] = 0;
+
+                  while (ioctl(precorder->dev_fd, AUDIOIOC_GETCAPS,
+                              (unsigned long)&cap) == cap.ac_len)
```
Comment (pkarashchenko) [inline system/nxrecorder/nxrecorder.c:187]:
```suggestion
                               (uintptr_t)&cap) == cap.ac_len)
```

File: system/nxrecorder/nxrecorder_main.c [hunk 5]
```diff
@@ -68,6 +68,8 @@ static int nxrecorder_cmd_quit(FAR struct nxrecorder_s *precorder,
                                FAR char *parg);
 static int nxrecorder_cmd_recordraw(FAR struct nxrecorder_s *precorder,
                                     FAR char *parg);
+static int nxrecorder_cmd_record(FAR struct nxrecorder_s *precorder,
+                                    FAR char *parg);
```
Comment (pkarashchenko) [inline system/nxrecorder/nxrecorder_main.c:72]:
```suggestion
static int nxrecorder_cmd_record(FAR struct nxrecorder_s *precorder,
                                 FAR char *parg);
```

File: system/nxrecorder/Kconfig (hunk 1 above)
Comment (qiaohaijiao) [reply system/nxrecorder/Kconfig]: resolved

File: system/nxrecorder/nxrecorder.c (hunk 2 above)
Comment (qiaohaijiao) [reply system/nxrecorder/nxrecorder.c]: resolved

File: system/nxrecorder/nxrecorder.c (hunk 3 above)
Comment (qiaohaijiao) [reply system/nxrecorder/nxrecorder.c]: resolved

File: system/nxrecorder/nxrecorder.c (hunk 4 above)
Comment (qiaohaijiao) [reply system/nxrecorder/nxrecorder.c]: resolved

File: system/nxrecorder/nxrecorder_main.c (hunk 5 above)
Comment (qiaohaijiao) [reply system/nxrecorder/nxrecorder_main.c]: resolved

File: system/nxrecorder/nxrecorder.c [hunk 6]
```diff
@@ -77,8 +133,13 @@
  *
  ****************************************************************************/
 
-static int nxrecorder_opendevice(FAR struct nxrecorder_s *precorder)
+static int nxrecorder_opendevice(FAR struct nxrecorder_s *precorder,
+                                 int format, int subfmt)
 {
+  struct audio_caps_s cap;
+  uint8_t supported = true;
```
Comment (pkarashchenko) [inline system/nxrecorder/nxrecorder.c:140]:
```suggestion
  bool supported = true;
```

**Review by pkarashchenko — COMMENTED** [review]
One minor comment. Other parts LGTM!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**